distage/distage-testkit-core/src/main/scala/izumi/distage/testkit/services/dstest/DistageTestRunner.scala

```diff
@@ -170,7 +160,8 @@ class DistageTestRunner[F[_] : TagK]
                   val allSharedKeys = mainSharedLocator.allInstances.map(_.key).toSet
 
                   val integrations = testplan.collectChildren[IntegrationCheck].map(_.target).toSet -- allSharedKeys
-                  val newtestplan = testInjector.trisectByRoots(appmodule.drop(allSharedKeys), testplan.keys -- allSharedKeys, integrations)
+                  val predicate: WeaknessPredicate = testplan.steps.map(_.target).contains
+                  val newtestplan = testInjector.trisectByRoots(appmodule.drop(allSharedKeys), testplan.keys -- allSharedKeys, integrations, predicate)
```

Comment:
The predicate should be applied in `shared`. Here it's no-op (non-weak keys that are referenced in test plan are already referenced in test plan, even without the predicate)

Reply:
Yeah, I found error in my tests.